Gems/Terrain/Code/Source/Components/TerrainWorldComponent.cpp

```diff
@@ -7,23 +7,70 @@
  */
 
 #include <Components/TerrainWorldComponent.h>
+#include <AzCore/Asset/AssetManager.h>
 #include <AzCore/Asset/AssetManagerBus.h>
 #include <AzCore/Component/Entity.h>
-#include <AzCore/Asset/AssetManager.h>
 #include <AzCore/RTTI/BehaviorContext.h>
 #include <AzCore/Serialization/EditContext.h>
 #include <AzCore/Serialization/SerializeContext.h>
 #include <AzFramework/Terrain/TerrainDataRequestBus.h>
 
 namespace Terrain
 {
+    
+    AZ::JsonSerializationResult::Result JsonTerrainWorldConfigSerializer::Load(
+        void* outputValue, [[maybe_unused]] const AZ::Uuid& outputValueTypeId,
+        const rapidjson::Value& inputValue, AZ::JsonDeserializerContext& context)
+    {
+        namespace JSR = AZ::JsonSerializationResult;
+
+        auto configInstance = reinterpret_cast<TerrainWorldConfig*>(outputValue);
+        AZ_Assert(configInstance, "Output value for JsonTerrainWorldConfigSerializer can't be null.");
+        
+        JSR::ResultCode result(JSR::Tasks::ReadField);
+        
+        result.Combine(ContinueLoadingFromJsonObjectField(
+            &configInstance->m_worldMin, azrtti_typeid<decltype(configInstance->m_worldMin)>(), inputValue, "WorldMin", context));
+        
+        result.Combine(ContinueLoadingFromJsonObjectField(
+            &configInstance->m_worldMax, azrtti_typeid<decltype(configInstance->m_worldMax)>(), inputValue, "WorldMax", context));
+
+        rapidjson::Value::ConstMemberIterator itr = inputValue.FindMember("HeightQueryResolution");
+        if (itr != inputValue.MemberEnd())
```

Comment:
This is not a big issue but I think we may need to combine the result with default used for debugging purpose, if the field is absent in the data file.
result.Combine(JSR::ResultCode(JSR::Tasks::ReadField, JSR::Outcomes::DefaultsUsed));

Reply:
If the field is absent in the data file, it'd just use the value from the default constructed object. Are there problems with that?